src/deepsparse/server/main.py

```diff
@@ -0,0 +1,283 @@
+            "Either define an alias or supply a single model for the task"
+        )
+
+
+def server_app_factory():
+    """
+    :return: a FastAPI app initialized with defined routes and ready for inference.
+        Use with a wsgi server such as uvicorn.
+    """
+    app = FastAPI(
+        title="deepsparse.server",
+        version=version,
+        description="DeepSparse Inference Server",
+    )
+    _LOGGER.info("created FastAPI app for inference serving")
+
+    config = server_config_from_env()
+    initialize_aysnc(config.workers)
+    _LOGGER.debug("loaded server config %s", config)
+    _add_general_routes(app, config)
+
+    pipeline_defs = load_pipelines_definitions(config)
+    _LOGGER.debug("loaded pipeline definitions from config %s", pipeline_defs)
+    num_tasks = len(config.models)
+    defined_tasks = set()
+    for pipeline_def in pipeline_defs:
+        _add_pipeline_route(app, pipeline_def, num_tasks, defined_tasks)
+
+    return app
+
+
+@click.command()
+@click.option(
+    "--host",
+    type=str,
+    default="0.0.0.0",
+    help="Bind socket to this host. Use --host 0.0.0.0 to make the application "
+    "available on your local network. "
+    "IPv6 addresses are supported, for example: --host '::'. Defaults to 0.0.0.0",
+)
+@click.option(
+    "--port",
+    type=int,
+    default=5543,
+    help="Bind to a socket with this port. Defaults to 5543.",
+)
+@click.option(
+    "--workers",
+    type=int,
+    default=1,
+    help="Use multiple worker processes. Defaults to 1.",
+)
+@click.option(
+    "--log_level",
+    type=str,
+    default="info",
+    help="Bind to a socket with this port. Defaults to info.",
+)
+@click.option(
+    "--config_file",
+    type=str,
+    default=None,
+    help="Configuration file containing info on how to serve the desired models.",
+)
+@click.option(
+    "--task",
+    type=str,
+    default=None,
+    help="The task the model_path is serving. For example, one of: "
+    "question_answering, text_classification, token_classification. "
+    "Ignored if config file is supplied",
+)
+@click.option(
+    "--model_path",
+    type=str,
+    default=None,
+    help="The path to a model.onnx file, a model folder containing the model.onnx "
+    "and supporting files, or a SparseZoo model stub. "
+    "Ignored if config_file is supplied.",
+)
+@click.option(
+    "--batch_size",
+    type=int,
+    default=1,
+    help="The batch size to serve the model from model_path with. "
+    "Ignored if config_file is supplied.",
+)
+def start_server(
+    host: str,
+    port: int,
+    workers: int,
+    log_level: str,
+    config_file: str,
+    task: str,
+    model_path: str,
+    batch_size: int,
+):
+    """
+    Start a DeepSparse inference server for serving the models and pipelines given
+    within the config_file or a single model defined by task, model_path, and batch_size
+
+    Example config.yaml for serving: \n
+        models: \n
```

Comment:
why are these `\n`s necessary here?

Reply:
click was automatically folding all of the text into single lines and ignoring the new lines. Found a work around in their docs now that uses \b before a paragraph to prevent this